class-pinterest-for-woocommerce.php

```diff
@@ -712,6 +725,54 @@ public static function maybe_disconnect_advertiser( $old_value, $new_value ) {
 			}
 		}
 
+
+		/**
+		 * Sync the Enhanced Match Support setting with Pinterest.
+		 *
+		 * @param array $old_value The old value of the option.
+		 * @param array $new_value The new value of the option.
+		 */
+		private static function maybe_sync_enhanced_match_support( $old_value, $new_value ) {
+
+			if (
+				! isset( $old_value['enhanced_match_support'] ) ||
+				! isset( $new_value['enhanced_match_support'] )
+			) {
+				return;
+			}
+
+			$tracking_tag = self::get_setting( 'tracking_tag', null );
+			if ( ! $tracking_tag ) {
+				return;
+			}
+
+			// Sync setting with Pinterest if old value is different than new ones.
+			if ( $old_value['enhanced_match_support'] !== $new_value['enhanced_match_support'] ) {
+				self::update_tag_config( $tracking_tag, $new_value['enhanced_match_support'] );
+			}
+		}
+
+		/**
+		 * Update the tag on Pinterest.
+		 *
+		 * @param string  $tracking_tag The connected tag.
+		 * @param boolean $enhanced_match_support The value of the enhanced_math_support option.
+		 */
+		public static function update_tag_config( $tracking_tag, $enhanced_match_support ) {
+			try {
+				Pinterest\API\Base::update_tag(
+					$tracking_tag,
+					array(
+						'aem_enabled' => $enhanced_match_support,
+					)
+				);
+
+			} catch ( \Exception $th ) {
```

Comment:
Minor: Variable name should `$e` for Exception, or `$th` for Throwable.

```suggestion
			} catch ( \Exception $e ) {
```